contracts/StakeManager.sol

```diff
@@ -157,12 +180,32 @@ contract StakeManager is Ownable {
                 revert StakeManager__InvalidLockTime();
             }
         }
+
         _mintBonusMP(account, deltaTime, _amount);
 
+        //mp estimation
+        uint256 mpPerEpoch = _getMPToMint(_amount, EPOCH_SIZE);
+        if(mpPerEpoch < 1){
+            revert StakeManager__StakeIsTooLow();
+        }
+        uint256 thisEpochExpiredMP = mpPerEpoch - _getMPToMint(_amount, epochEnd() - block.timestamp);
```

Comment:
@3esmit I noticed that this might be wrong here.

The missed MP should be

```
mpPerEpoch - _getMPToMint(_amount, block.timestamp - currentEpoch.startTime);
```

We want to keep track of the MP we've *missed* so the time from beginning of epoch til now.
Not the MP that are yet to come (now till current epoch end).

Can you confirm this?

Reply:
The code is correct. I had to make this way to keep the same math as in the processAccount. 
The past commit used to do `block.timestamp - epochs[currentEpoch].startTime`, which would get us the amount of MPs that already expired in currentEpoch. However, due rounding errors, I had to change it to use the exact same calculation in the processAccount, so instead of doing that, I calculate how much the account would get in current epoch, and reduce this value from the mpPerEpoch, resulting in the amount of MPs that expired. 

This seems counter intuitive, however is needed only due solidity/evm nature of math.